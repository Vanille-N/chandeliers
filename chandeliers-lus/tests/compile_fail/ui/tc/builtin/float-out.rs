//! Wrong output type.
chandeliers_lus::decl! {
    extern node float_of_int(i : int) returns (f : float);

    node main() returns ();
    var x, y : float;
    let
        (x, y) = float_of_int(1);
    tel
}

fn main() {}
