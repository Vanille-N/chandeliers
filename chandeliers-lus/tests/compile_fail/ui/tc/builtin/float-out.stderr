error: Type mismatch between the left and right sides: expected a tuple (float, float), got a scalar float
 --> tests/fail/ui/tc/builtin/float-out.rs:7:9
  |
7 |         (x, y) = float_of_int(1);
  |         ^^^^^^^^^^^^^^^^^^^^^^^^
  |
note: This element has type (float, float)
 --> tests/fail/ui/tc/builtin/float-out.rs:7:9
  |
7 |         (x, y) = float_of_int(1);
  |         ^^^^^
note: While this element has type float
 --> tests/fail/ui/tc/builtin/float-out.rs:7:18
  |
7 |         (x, y) = float_of_int(1);
  |                  ^^^^^^^^^^^^^^^
